build/elasticsearch/bin/es-docker

```diff
@@ -22,4 +22,6 @@ do
     fi
 done < <(env)
 
+echo "-Des.cgroups.hierarchy.override=/" >> config/jvm.options
```

Comment:
> Correct me if I'm wrong please, but I don't think we can do it in the Dockerfile since that bakes the value of ES_JAVA_OPTS into the image but users can override this externally when running the image.

Yes that is correct and in the website docs [docker-compose.yml](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html#docker-cli-run-prod-mode) example we are showing how to use `ES_JAVA_OPTS` as one method of defining the jvm heap size.

@jasontedor Just so that I understand better this solution, basically `es.cgroups.hierarchy.override` can be defined as a jvm option, as documented [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/setting-system-settings.html#jvm-options) hence will be honored when presented through `ES_JAVA_OPTS`?
I am asking this as [my understanding](https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html#_setting_default_settings) is that most parameters defined in `jvm.options` can also be passed as `-E` style cli args. If that is true, the user can run the image using `docker run ... -e es.cgroups.hierarchy.override=/` and so it could be the value [here](https://github.com/elastic/elasticsearch-docker/blob/master/build/elasticsearch/bin/es-docker#L11) instead of the existing `es_opts=''`?

(Note that I tried the last point with 6.0.0-alpha1-SNAPSHOT and received:
```
uncaught exception in thread [main]
org.elasticsearch.bootstrap.StartupException: java.lang.IllegalArgumentException: unknown setting [es.cgroups.hierarchy.override] please check that any required plugins are installed, or check the breaking changes documentation for removed settings
```
)